rustls/src/msgs/handshake.rs

```diff
@@ -570,7 +570,65 @@ impl TlsListElement for KeyShareEntry {
     const SIZE_LEN: ListLength = ListLength::U16;
 }
 
+/// The body of the `SupportedVersions` extension when it appears in a
+/// `ClientHello`
+///
+/// This is documented as a preference-order vector, but we (as a server)
+/// ignore the preference of the client.
+///
 /// RFC8446: `ProtocolVersion versions<2..254>;`
+#[derive(Clone, Copy, Debug, Default)]
+pub struct SupportedProtocolVersions {
+    pub(crate) tls13: bool,
+    pub(crate) tls12: bool,
+}
+
+impl SupportedProtocolVersions {
+    pub(crate) fn any(&self, func: impl Fn(ProtocolVersion) -> bool) -> bool {
```

Comment:
WDYT about calling the `func` callback something like `filter` or `allowed` instead? I think it's a bit clearer.

Reply:
Though since the stdlib [uses `f`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) for this, I can be convinced it doesn't matter at all :-)